src/index.ts

```diff
@@ -357,7 +357,7 @@ function ${cls.name}${
 		});
 	}
 
-	const staticFields = cls.static_properties?.map(field => `${field.name} = ${field.value}`).join(", ") ?? "";
+	const staticFields = (cls.static_properties?.length? `\n${cls.static_properties.map(field => `${cls.name}.${field.name} = ${field.value}`).join("\n")}` : "") ?? "";
```

Comment:
nit: The nullish coalescing operator (`??`) is redundant here as the ternary is always defined and non-null. Small formatting change (the ? after .length is different to `?.`, which is optional chaining)
```suggestion
	const staticFields = cls.static_properties?.length ? `\n${cls.static_properties.map(field => `${cls.name}.${field.name} = ${field.value}`).join("\n")}` : "";
```